bctl/daemon/mrtap/bzcert/zliconfig/zliconfig.go

```
@@ -77,40 +97,179 @@ func (z *ZLIConfig) Load() error {
+	} else if err := json.Unmarshal(data, &z.idpType); err != nil {
+		return fmt.Errorf("malformed mrtap idp: %w", err)
+	}
+
+	if data, err := db.Get([]byte("clientId"), nil); err != nil || len(data) == 0 {
+		return fmt.Errorf("failed to retrieve clientId: %w", err)
+	} else {
+		z.clientId = string(data[:])
+	}
+
+	// clientSecret is optional per IdP
+	if data, err := db.Get([]byte("clientSecret"), nil); err == nil {
+		z.clientSecret = string(data[:])
+	}
+
+	if data, err := db.Get([]byte("sessionId"), nil); err != nil || len(data) == 0 {
+		return fmt.Errorf("failed to retrieve sessionId: %w", err)
+	} else {
+		z.sessionId = string(data[:])
+	}
+
+	if data, err := db.Get([]byte("sessionToken"), nil); err != nil || len(data) == 0 {
+		return fmt.Errorf("failed to retrieve sessionToken: %w", err)
+	} else {
+		z.sessionToken = string(data[:])
 	}
 
 	return nil
 }
 
-func (z *ZLIConfig) Refresh() error {
-	if z.refreshCommand == "" {
-		return fmt.Errorf("could not refresh zli config, because no refresh command was found")
+// This function expects to be passed a db connection that has already been opened
+func (z *ZLIConfig) setTokenRefresh(db *leveldb.DB) (err error) {
+
+	if data, err := json.Marshal(z.TokenSet); err != nil {
+		return fmt.Errorf("malformed tokenSet config: %w", err)
+	} else if err = db.Put([]byte("tokenSet"), data, nil); err != nil {
+		return fmt.Errorf("failed to set tokenSet config: %w", err)
+	}
+
+	if err = db.Put([]byte("sessionId"), []byte(z.sessionId), nil); err != nil {
+		return fmt.Errorf("failed to set sessionId config: %w", err)
 	}
 
-	// Update the id token by calling the passed in zli refresh command
-	if err := runRefreshCommand(z.refreshCommand); err != nil {
+	if err = db.Put([]byte("sessionToken"), []byte(z.sessionToken), nil); err != nil {
+		return fmt.Errorf("failed to set sessionId config: %w", err)
+	}
+
+	return nil
+}
+
+func (z *ZLIConfig) Refresh() (err error) {
+	var db *leveldb.DB
```

Should we put a check here now well so that we can return an error if a SA daemon attempts to refresh:

```go
if z.WhoAmI.Type != "User" {
		return fmt.Errorf("subject of type %s is not able to perform IdP refresh", z.WhoAmI.Type)
	}
```